src/Symfony/Component/Cache/Adapter/RedisTagAwareAdapter.php

```diff
@@ -305,4 +306,193 @@ private function getRedisEvictionPolicy(): string
 
         return $this->redisEvictionPolicy = '';
     }
+
+
+    protected function getPrefix(): string
+    {
+        if ($this->redis instanceof \Predis\ClientInterface) {
+            $prefix = $this->redis->getOptions()->prefix ? $this->redis->getOptions()->prefix->getPrefix() : '';
+        } elseif (\is_array($prefix = $this->redis->getOption(\Redis::OPT_PREFIX) ?? '')) {
+            $prefix = current($prefix);
+        }
+        return $prefix;
+    }
+
+    /**
+     * Returns all existing tag keys from the cache.
+     *
+     * @TODO Verify the LUA scripts are redis-cluster safe.
+     *
+     * @return array
+     */
+    protected function getAllTagKeys(): array
+    {
+        $tagKeys = [];
+        $prefix = $this->getPrefix();
+        // need to trim the \0 for lua script
+        $tagsPrefix = trim(self::TAGS_PREFIX);
+
+        // get all SET entries which are tagged
+        $getTagsLua = <<<'EOLUA'
+            redis.replicate_commands()
+            local cursor = ARGV[1]
+            local prefix = ARGV[2]
+            local tagPrefix = string.gsub(KEYS[1], prefix, "")
+            return redis.call('SCAN', cursor, 'COUNT', 5000, 'MATCH', '*' .. tagPrefix .. '*', 'TYPE', 'set')
+        EOLUA;
+        $cursor = null;
+        do {
+            $results = $this->pipeline(function () use ($getTagsLua, $cursor, $prefix, $tagsPrefix) {
+                yield 'eval' => [$getTagsLua, [$tagsPrefix, $cursor, $prefix], 1];
+            });
+
+            $setKeys = $results->valid() ? iterator_to_array($results) : [];
+            [$cursor, $ids] = $setKeys[$tagsPrefix] ?? [null, null];
+            // merge the fetched ids together
+            $tagKeys = array_merge($tagKeys, $ids);
+        } while ($cursor = (int) $cursor);
+
+        return $tagKeys;
+    }
+
+
+    /**
+     * Checks all tags in the cache for orphaned items and creates a "report" array.
+     *
+     * By default, only completely orphaned tag keys are reported. If
+     * compressMode is enabled the report will include all orphaned cache item
+     * reference for the
+     *
+     * @TODO Verify the LUA scripts are redis-cluster safe.
+     * @TODO Is there anything that can be done to reduce memory footprint?
+     *
+     * @param bool $compressMode
+     * @return array{tagKeys: string[], orphanedTagKeys: string[], orphanedTagReferenceKeys?: array<string, string[]>}
+     *      tagKeys: List of all tags in the cache.
+     *      orphanedTagKeys: List of tags that only reference orphaned cache items.
+     *      orphanedTagReferenceKeys: List of all orphaned cache item references per tag.
+     *                           Keyed by tag, value is the list of orphaned cache item keys.
+     */
+    protected function getOrphanedTagsStats(bool $compressMode = false): array
```

Comment:
why is this protected ?

Reply:
protected does not allow that. A Console or UI would need to have a public method.